### PR TITLE
Docs

### PR DESCRIPTION
**Pull request by lordofthejars**
Adds new configuration properties for affected strategy and adds in refcard section, a reference for each configuration property of each strategy.

### CODE REVIEW

File: docs/configuration.adoc [hunk 1]
```diff
@@ -68,9 +68,44 @@ TIP: `git.commit`, `git.previous.commit` and `git.last.commits` Java system prop
 When this strategy gets all changes then inspect all tests of current project checking which ones _imports_ these classes.
 If the test exercises a business class that has been modified, we treat it as *important*  so it will be executed earlier in the test plan.
 
+.About transitivity
+****
+Currently any we are resolving in transitive way imports.
```
Comment (bartoszmajsak) [inline docs/configuration.adoc]: any what? I think we have to work on this sentence.

File: docs/configuration.adoc [hunk 2]
```diff
@@ -68,9 +68,44 @@ TIP: `git.commit`, `git.previous.commit` and `git.last.commits` Java system prop
...
+This means that any import of a business class is considered as a dependency of the test too.
+Let's look an example:
```
Comment (bartoszmajsak) [inline docs/configuration.adoc]: at the example?

File: docs/configuration.adoc [hunk 3]
```diff
@@ -68,9 +68,44 @@ TIP: `git.commit`, `git.previous.commit` and `git.last.commits` Java system prop
...
+
+Suppose we have `ATest.java` which imports `A.java`.
+At the same time `A.java` imports `B.java` (ATest -> A -> B).
+If `B.java` is modified, then `ATest.java` is considered an *important* test too.
+
+By default this import transitivity is applied to all imports except the ones from `java` and `javax`.
+
+Sometimes you might want to stop this transitivity before reaching these imports, for example in case of developing an application with Spring, you'd probably want to *exclude* `org.springframework` imports.
+Or maybe just *include* imports from your business code, for example all imports from `org.superbiz`.
```
Comment (bartoszmajsak) [inline docs/configuration.adoc]: I would avoid any particular library. Let's make sth up. The point is that we use additional libraries which won't change - so we want to optimize it.

File: docs/configuration.adoc [hunk 4]
```diff
@@ -68,9 +68,44 @@ TIP: `git.commit`, `git.previous.commit` and `git.last.commits` Java system prop
...
+----
+
+INFO: Exclusions has precedence over inclusions.
+
+****
+
 IMPORTANT: This strategy is currently only applicable for _white box_ testing approach. At this point our approach is to
  analyze direct code dependencies, but we are working on broader use cases.
 
+WARNING: This strategy does not work with Java 9.
```
Comment (bartoszmajsak) [inline docs/configuration.adoc]: "At this moment this strategy does not work with Java 9." ?

File: docs/what-is-this.adoc [hunk 5]
```diff
@@ -28,7 +28,7 @@ and `maven-failsafe-plugin`.
 
 === Minimum requirements
 
-Java:: Requires Java 8.
+Java:: Requires Java 8. In case of using <<Affected>> strategy, Java 9 is not supported yet.
```
Comment (dipak-pawar) [inline docs/what-is-this.adoc]: This looks little confusing. Are we supporting java9 for other strategies. I don't remember anyone tested smart-testing tool with java9

File: docs/refcard.adoc [hunk 6]
```diff
@@ -35,3 +35,41 @@ a|`smart.testing.apply.to`
 a|`surefire`, `failsafe`
 a|`surefire, failsafe`
 |===
+
+=== Strategies
```
Comment (dipak-pawar) [inline docs/refcard.adoc]: Can we give change this? I believe following table has more information than strategies

Comment (lordofthejars) [inline docs/refcard.adoc]: No, it only contains properties that only configures strategies things

File: docs/what-is-this.adoc (hunk 5 above)
Comment (lordofthejars) [inline docs/what-is-this.adoc]: @bartoszmajsak did and found not working `affected` because of JavaAssist

Comment (bartoszmajsak) [inline docs/what-is-this.adoc]: Sure, but I haven't checked the other. Something to investigate deeper.

Comment (lordofthejars) [inline docs/what-is-this.adoc]: I think that for the others it should work because we are not manipulating the `.class` but something to take a  look.

**Pull request merged**